dotnet/samples/KernelSyntaxExamples/Example65_HandlebarsPlanner.cs

```diff
@@ -126,7 +126,9 @@ private static async Task PlanNotPossibleSampleAsync(bool shouldPrintPrompt = fa
             // Load additional plugins to enable planner but not enough for the given goal.
             await RunSampleAsync("Send Mary an email with the list of meetings I have scheduled today.", shouldPrintPrompt, "SummarizePlugin");
         }
-        catch (KernelException e)
+        catch (KernelException ex) when (
+            ex.Message.Contains("The plan references hallucinated helpers", StringComparison.CurrentCultureIgnoreCase)
+            || ex.Message.Contains("Unable to create plan for goal with available functions", StringComparison.CurrentCultureIgnoreCase))
```

Comment:
We generally discourage developers from parsing exception messages. It's very fragile, both to tweaks / improvements over time and to localization (improving an exception message could break such code). If it's important to be able to differentiate these cases, we should consider throwing a more derived exception type and/or differentiating these cases with an enum error code on the exception.